### ISSUE TITLE
OpenAPI v3.1 Support

### ISSUE THREAD

**Issue by philsturgeon** [oasis-tcs/odata-openapi#207]
Thanks for making this excellent package. I wanted to make a quick issue to track the progress towards supporting OpenAPI v3.1, because it's been out for over a year now and it's excellent. It adds a lot of convenient functionality and lines up perfectly with JSON Schema, which could actually simplify your toolchain rather well as you can just use any of the vast amounts of JSON Schema tooling instead of the subset of tooling which understands what an OpenAPI Schema object.

Here's a guide to the differences written for end-users, but it should help make things fairly clear to tooling vendors too. 

https://www.openapis.org/blog/2021/02/16/migrating-from-openapi-3-0-to-3-1-0

If you say you'll be working on it I can update [openapi.tools](https://openapi.tools) to reflect that. 

Let me know if you have any questions or need any help with the upgrade process.

**Comment by ralfhandl** [comment]
Hi Phil,

Thanks for the reminder!

Do you happen to know when Swagger UI will support OpenAPI v3.1? Or would you recommend a different UI tool now?

Thanks in advance

**Comment by philsturgeon** [comment]
I don't know when/if Swagger UI will support OAS 3.1, but there are loads of alternatives on https://openapi.tools/.

I personally recommend [Stoplight Elements](https://github.com/stoplightio/elements) (becuase I helped build it and happen to think its really good).

**Comment by ralfhandl** [comment]
Thanks, Stoplight Elements really looks nice!

**Comment by ralfhandl** [comment]
Hi Phil,

Looking at your helpful guide https://www.openapis.org/blog/2021/02/16/migrating-from-openapi-3-0-to-3-1-0 there really doesn't seem that much to do.

One thing I wonder how to rephrase in v3.1 is "nullable references":
~~~json
"properties": {
    "Superordinate": {
        "allOf": [
            {
                "$ref": "#/components/schemas/org.example.odata.salesservice.SalesOrganization"
            }
        ],
        "nullable": true
    },
~~~
This means that the property `Superordinate` is either null or has a value defined by the referenced schema.

Do I correctly assume that I have to rephrase this as
~~~json
"properties": {
    "Superordinate": {
        "anyOf": [
            {
                "$ref": "#/components/schemas/org.example.odata.salesservice.SalesOrganization"
            },
            {
                "type": "null"
            },
        ]
    },
~~~

Thanks in advance

**Comment by ralfhandl** [comment]
Hi @philsturgeon,

This tool now supports OpenAPI v3.1, and I have created the PR https://github.com/apisyouwonthate/openapi.tools/pull/313 to have your tools list updated accordingly.

Thanks in advance!

**Comment by philsturgeon** [comment]
This is amazing, thank you so much for getting on top of that! And in short yeah, I believe your example in the above comment to be valid.
